odoo/fields.py

```diff
@@ -893,10 +893,19 @@ def update_db_notnull(self, model, column):
             if model._table_has_rows():
                 model._init_column(self.name)
 
+        err_key = (model._table, self.name)
         if self.required and not has_notnull:
-            sql.set_not_null(model._cr, model._table, self.name)
+            err = sql.set_not_null(model._cr, model._table, self.name)
+            if err:
+                model.pool._notnull_errors[err_key] = err
```

Comment:
This makes a strong connection between the output of `set_not_null` and the internals of the registry.
I suggest to make `set_not_null` return the exception, and to only put this exception in `registry._notnull_errors`.